{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Notebook for generating master tables\n",
    "\n",
    "- Combining LFC data, Z-score normalization, editing information into a single dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import seaborn as sns\n",
    "import warnings\n",
    "import os\n",
    "import functools\n",
    "import upsetplot\n",
    "\n",
    "warnings.filterwarnings('ignore')\n",
    "plt.rc('font', family='Helvetica')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "MBES = pd.read_csv('MBESv2_CORRECTED.csv')\n",
    "\n",
    "#adding in gene names\n",
    "genes = []\n",
    "for i, val in MBES.iterrows():\n",
    "    t = val['classification']\n",
    "    if t=='safe-targeting control':\n",
    "        genes.append('ST')\n",
    "    elif t=='non-targeting control':\n",
    "        genes.append('NT')\n",
    "    elif t=='targeting guide':\n",
    "        g = val['gene_name_m']\n",
    "        if str(g)=='nan':\n",
    "            g = val['Symbol']\n",
    "\n",
    "        genes.append(g)\n",
    "\n",
    "MBES['gene_name_m_corrected'] = genes\n",
    "\n",
    "#and info about whether its a legacy guides\n",
    "legacy = []\n",
    "for i in list(MBES['mutation_idx']):\n",
    "    if 'MBESv1_legacy' in str(i):\n",
    "        legacy.append(True)\n",
    "    else:\n",
    "        legacy.append(False)\n",
    "\n",
    "MBES['legacy'] = legacy\n",
    "\n",
    "#subsetting\n",
    "ABE = MBES[MBES['Editor']=='ABE']\n",
    "CBE = MBES[MBES['Editor']=='CBE']\n",
    "\n",
    "#and then add back in the non-targetting guides\n",
    "nt_ABE = list(ABE[ABE['classification']=='non-targeting control']['gRNA_id'])\n",
    "nt_CBE = list(CBE[CBE['classification']=='non-targeting control']['gRNA_id'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "ABE_mageck = ['ABE_d5_input.sgrna_summary.txt',\n",
    " 'ABE_d15_input.sgrna_summary.txt',\n",
    " 'ABE_bonemarrow_input.sgrna_summary.txt',\n",
    " 'ABE_spleen_input.sgrna_summary.txt',\n",
    " 'ABE_meninges_input.sgrna_summary.txt',\n",
    " ]\n",
    "\n",
    "CBE_mageck = ['CBE_d5_input.sgrna_summary.txt',\n",
    " 'CBE_d15_input.sgrna_summary.txt',\n",
    " 'CBE_bonemarrow_input.sgrna_summary.txt',\n",
    " 'CBE_spleen_input.sgrna_summary.txt',\n",
    " 'CBE_meninges_input.sgrna_summary.txt',\n",
    " ]\n",
    "\n",
    "names = ['In Vitro D5','In Vitro D15','Bone', 'Spleen', 'Meninges',]\n",
    "\n",
    "abe_df_holder = []\n",
    "cbe_df_holder = []\n",
    "for i in ABE_mageck:\n",
    "    d = pd.read_csv(f'mageck/sgrna_summaries/{i}', sep='\\t')\n",
    "    abe_df_holder.append(d)\n",
    "for i in CBE_mageck:\n",
    "    d = pd.read_csv(f'mageck/sgrna_summaries/{i}', sep='\\t')\n",
    "    cbe_df_holder.append(d)\n",
    "\n",
    "ABE_mageck_dict = dict(zip(names, abe_df_holder))\n",
    "CBE_mageck_dict = dict(zip(names, cbe_df_holder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#FIRST ABE\n",
    "samples = ['In Vitro D15','Bone', 'Spleen', 'Meninges',]\n",
    "\n",
    "excluded_guides = []\n",
    "df_holder = []\n",
    "\n",
    "\n",
    "for samp in samples:\n",
    "    mageck_df = ABE_mageck_dict[samp]\n",
    "    mageck_df = mageck_df.rename(columns = {'sgrna':'gRNA_id', 'LFC':f'LFC_{samp}', 'treat_mean':f'treat_mean_{samp}', 'control_mean':f'control_mean_{samp}',\n",
    "                                 'treatment_count':f'treat_count_{samp}', 'control_count':f'control_count_{samp}', 'FDR':f'FDR_{samp}'})\n",
    "\n",
    "    #exc = list(mageck_df[mageck_df[f'control_mean_{samp}']<control_mean_minimum]['gRNA_id'])\n",
    "    #for i in exc:\n",
    "    #    excluded_guides.append(i)\n",
    "    #include everything and filter afterward instead\n",
    "\n",
    "\n",
    "    df_holder.append(mageck_df[['gRNA_id', f'LFC_{samp}', f'treat_mean_{samp}', f'control_mean_{samp}',f'treat_count_{samp}', f'control_count_{samp}', f'FDR_{samp}']])\n",
    "    #mageck_df = mageck_df[mageck_df['control_mean']>=control_mean_minimum].sort_values(by='LFC', ascending=False)\n",
    "    \n",
    "df_merged = functools.reduce(lambda  left,right: pd.merge(left,right,on=['gRNA_id'],\n",
    "                                            how='outer'), df_holder)\n",
    "\n",
    "\n",
    "#and then excluder the guides with too few control counts\n",
    "exc_g = list(np.unique(excluded_guides))\n",
    "df_merged = df_merged[~df_merged['gRNA_id'].isin(exc_g)]\n",
    "\n",
    "\n",
    "#and then calculate the z-score for each\n",
    "for k in samples:\n",
    "    lfcs = np.asarray(df_merged[f'LFC_{k}'])\n",
    "    avg = np.average(lfcs)\n",
    "    std = np.std(lfcs)\n",
    "\n",
    "    zscore = (lfcs-avg)/std\n",
    "    df_merged[f'Z_{k}'] = zscore\n",
    "\n",
    "df_merged_ABE = df_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Then CBE\n",
    "\n",
    "samples = ['In Vitro D15','Bone', 'Spleen', 'Meninges',]\n",
    "\n",
    "excluded_guides = []\n",
    "df_holder = []\n",
    "\n",
    "\n",
    "for samp in samples:\n",
    "    mageck_df = CBE_mageck_dict[samp]\n",
    "    mageck_df = mageck_df.rename(columns = {'sgrna':'gRNA_id', 'LFC':f'LFC_{samp}', 'treat_mean':f'treat_mean_{samp}', 'control_mean':f'control_mean_{samp}',\n",
    "                                 'treatment_count':f'treat_count_{samp}', 'control_count':f'control_count_{samp}', 'FDR':f'FDR_{samp}'})\n",
    "\n",
    "    #exc = list(mageck_df[mageck_df[f'control_mean_{samp}']<control_mean_minimum]['gRNA_id'])\n",
    "    #for i in exc:\n",
    "    #    excluded_guides.append(i)\n",
    "    #include everything and filter afterward instead\n",
    "\n",
    "\n",
    "    df_holder.append(mageck_df[['gRNA_id', f'LFC_{samp}', f'treat_mean_{samp}', f'control_mean_{samp}',f'treat_count_{samp}', f'control_count_{samp}', f'FDR_{samp}']])\n",
    "    #mageck_df = mageck_df[mageck_df['control_mean']>=control_mean_minimum].sort_values(by='LFC', ascending=False)\n",
    "    \n",
    "df_merged = functools.reduce(lambda  left,right: pd.merge(left,right,on=['gRNA_id'],\n",
    "                                            how='outer'), df_holder)\n",
    "\n",
    "\n",
    "#and then excluder the guides with too few control counts\n",
    "exc_g = list(np.unique(excluded_guides))\n",
    "df_merged = df_merged[~df_merged['gRNA_id'].isin(exc_g)]\n",
    "\n",
    "\n",
    "#and then calculate the z-score for each\n",
    "for k in samples:\n",
    "    lfcs = np.asarray(df_merged[f'LFC_{k}'])\n",
    "    avg = np.average(lfcs)\n",
    "    std = np.std(lfcs)\n",
    "\n",
    "    zscore = (lfcs-avg)/std\n",
    "    df_merged[f'Z_{k}'] = zscore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#USING THE D5 EDITING AS PROXY FOR ALL OTHER EDITING\n",
    "\n",
    "d5_abe = pd.read_csv('ABE_editing/MLE/d5_ABE_MLE.csv')\n",
    "##---USING UPDATED CBE EDITING THAT INCLUDES LEGACY GUIDES!!---\n",
    "d5_cbe = pd.read_csv('CBE_editing_UPDATED/MLE/d5_CBE_MLE.csv')\n",
    "\n",
    "#and then adding in editing information to the dataframe\n",
    "abe_editing = d5_abe[['Guide_ID', 'Reads_aligned_all_amplicons', 'corr_perc',\t'target_base_edit_perc', 'byproduct_INDEL_perc','byproduct_sub_perc']].rename(columns = {'Guide_ID':'gRNA_id', 'Reads_aligned_all_amplicons': 'sensor_reads'})\n",
    "abe_editing = abe_editing[abe_editing['gRNA_id'].isin(df_merged_ABE['gRNA_id'])]\n",
    "\n",
    "cbe_editing = d5_cbe[['Guide_ID', 'Reads_aligned_all_amplicons', 'corr_perc',\t'target_base_edit_perc', 'byproduct_INDEL_perc','byproduct_sub_perc']].rename(columns = {'Guide_ID':'gRNA_id', 'Reads_aligned_all_amplicons': 'sensor_reads'})\n",
    "cbe_editing = cbe_editing[cbe_editing['gRNA_id'].isin(df_merged['gRNA_id'])]\n",
    "\n",
    "ABE_master = pd.merge(df_merged_ABE, abe_editing, on='gRNA_id')\n",
    "CBE_master = pd.merge(df_merged, cbe_editing, on='gRNA_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_master = pd.concat((ABE_master, CBE_master))\n",
    "combined_master_final = pd.merge(MBES, combined_master, on='gRNA_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_master_final_control_50 = combined_master_final[combined_master_final['control_mean_In Vitro D15']>=50]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_master_final.to_csv('master_table_unfiltered.csv', index=False)\n",
    "combined_master_final_control_50.to_csv('master_table_min50_control_mean_invitro.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# OncoKB annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "oncokb_annots = pd.read_exc el('source_data/oncokb_annotations.xlsx', header=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>LEVEL_3B</th>\n",
       "      <th>LEVEL_4</th>\n",
       "      <th>LEVEL_R1</th>\n",
       "      <th>LEVEL_R2</th>\n",
       "      <th>LEVEL_R3</th>\n",
       "      <th>HIGHEST_LEVEL</th>\n",
       "      <th>tcn</th>\n",
       "      <th>lcn</th>\n",
       "      <th>purity</th>\n",
       "      <th>clonality</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32930688</td>\n",
       "      <td>32930688</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.328395</td>\n",
       "      <td>SUBCLONAL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SETD2</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>3</td>\n",
       "      <td>47103692</td>\n",
       "      <td>47103692</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.328395</td>\n",
       "      <td>SUBCLONAL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TERT</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295228</td>\n",
       "      <td>+</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.328395</td>\n",
       "      <td>CLONAL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DAXX</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>6</td>\n",
       "      <td>33289023</td>\n",
       "      <td>33289023</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.328395</td>\n",
       "      <td>SUBCLONAL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3CG</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>7</td>\n",
       "      <td>106524667</td>\n",
       "      <td>106524667</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.328395</td>\n",
       "      <td>SUBCLONAL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88405</th>\n",
       "      <td>NPM1P42</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>15</td>\n",
       "      <td>73191893</td>\n",
       "      <td>73191893</td>\n",
       "      <td>+</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>RNA</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88406</th>\n",
       "      <td>RP11-432M8.17</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>5</td>\n",
       "      <td>17655254</td>\n",
       "      <td>17655254</td>\n",
       "      <td>+</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>RNA</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88407</th>\n",
       "      <td>TOP2A</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>38567159</td>\n",
       "      <td>38567159</td>\n",
       "      <td>+</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Intron</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88408</th>\n",
       "      <td>ENSR00000186023</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>20</td>\n",
       "      <td>48626673</td>\n",
       "      <td>48626673</td>\n",
       "      <td>+</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>IGR</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88409</th>\n",
       "      <td>C4orf22</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>4</td>\n",
       "      <td>81299052</td>\n",
       "      <td>81299052</td>\n",
       "      <td>+</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Intron</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>88410 rows Ã— 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Hugo_Symbol Center NCBI_Build Chromosome  Start_Position  \\\n",
       "0                BRCA2  MSKCC     GRCh37         13        32930688   \n",
       "1                SETD2  MSKCC     GRCh37          3        47103692   \n",
       "2                 TERT  MSKCC     GRCh37          5         1295228   \n",
       "3                 DAXX  MSKCC     GRCh37          6        33289023   \n",
       "4               PIK3CG  MSKCC     GRCh37          7       106524667   \n",
       "...                ...    ...        ...        ...             ...   \n",
       "88405          NPM1P42  MSKCC     GRCh37         15        73191893   \n",
       "88406    RP11-432M8.17  MSKCC     GRCh37          5        17655254   \n",
       "88407            TOP2A  MSKCC     GRCh37         17        38567159   \n",
       "88408  ENSR00000186023  MSKCC     GRCh37         20        48626673   \n",
       "88409          C4orf22  MSKCC     GRCh37          4        81299052   \n",
       "\n",
       "       End_Position Strand                         Consequence  \\\n",
       "0          32930688      +                    missense_variant   \n",
       "1          47103692      +                    missense_variant   \n",
       "2           1295228      +               upstream_gene_variant   \n",
       "3          33289023      +                    missense_variant   \n",
       "4         106524667      +                    missense_variant   \n",
       "...             ...    ...                                 ...   \n",
       "88405      73191893      +  non_coding_transcript_exon_variant   \n",
       "88406      17655254      +  non_coding_transcript_exon_variant   \n",
       "88407      38567159      +                      intron_variant   \n",
       "88408      48626673      +           regulatory_region_variant   \n",
       "88409      81299052      +                      intron_variant   \n",
       "\n",
       "      Variant_Classification Variant_Type  ... LEVEL_3B LEVEL_4 LEVEL_R1  \\\n",
       "0          Missense_Mutation          SNP  ...      NaN     NaN      NaN   \n",
       "1          Missense_Mutation          SNP  ...      NaN     NaN      NaN   \n",
       "2                    5'Flank          SNP  ...      NaN     NaN      NaN   \n",
       "3          Missense_Mutation          SNP  ...      NaN     NaN      NaN   \n",
       "4          Missense_Mutation          SNP  ...      NaN     NaN      NaN   \n",
       "...                      ...          ...  ...      ...     ...      ...   \n",
       "88405                    RNA          SNP  ...      NaN     NaN      NaN   \n",
       "88406                    RNA          SNP  ...      NaN     NaN      NaN   \n",
       "88407                 Intron          SNP  ...      NaN     NaN      NaN   \n",
       "88408                    IGR          SNP  ...      NaN     NaN      NaN   \n",
       "88409                 Intron          SNP  ...      NaN     NaN      NaN   \n",
       "\n",
       "      LEVEL_R2  LEVEL_R3  HIGHEST_LEVEL  tcn  lcn    purity  clonality  \n",
       "0          NaN       NaN            NaN  4.0  1.0  0.328395  SUBCLONAL  \n",
       "1          NaN       NaN            NaN  3.0  1.0  0.328395  SUBCLONAL  \n",
       "2          NaN       NaN            NaN  5.0  NaN  0.328395     CLONAL  \n",
       "3          NaN       NaN            NaN  4.0  1.0  0.328395  SUBCLONAL  \n",
       "4          NaN       NaN            NaN  6.0  0.0  0.328395  SUBCLONAL  \n",
       "...        ...       ...            ...  ...  ...       ...        ...  \n",
       "88405      NaN       NaN            NaN  NaN  NaN       NaN        NaN  \n",
       "88406      NaN       NaN            NaN  NaN  NaN       NaN        NaN  \n",
       "88407      NaN       NaN            NaN  NaN  NaN       NaN        NaN  \n",
       "88408      NaN       NaN            NaN  NaN  NaN       NaN        NaN  \n",
       "88409      NaN       NaN            NaN  NaN  NaN       NaN        NaN  \n",
       "\n",
       "[88410 rows x 36 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "oncokb_annots[['Hugo_Symbol', 'Center', 'NCBI_Build', 'Chromosome', 'Start_Position',\n",
    "       'End_Position', 'Strand', 'Consequence', 'Variant_Classification',\n",
    "       'Variant_Type', 'Reference_Allele', 'Tumor_Seq_Allele1',\n",
    "       'Tumor_Seq_Allele2', 'Tumor_Sample_Barcode', 't_ref_count',\n",
    "       't_alt_count', 'n_ref_count', 'n_alt_count', 'Transcript_ID',\n",
    "       'HGVSp_Short', 'RefSeq', 'HGVSc', 'ONCOGENIC', 'LEVEL_1', 'LEVEL_2',\n",
    "       'LEVEL_3A', 'LEVEL_3B', 'LEVEL_4', 'LEVEL_R1', 'LEVEL_R2', 'LEVEL_R3',\n",
    "       'HIGHEST_LEVEL', 'tcn', 'lcn', 'purity', 'clonality',]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "MBES = pd.read_csv('MBESv2_CORRECTED.csv')\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
